ipn/ipnlocal/c2n.go

```diff
@@ -139,6 +147,25 @@ func (b *LocalBackend) handleC2N(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// handleC2NAppConnectorDomainRoutesGet handles returning the domains
+// that the app connector is responsible for, as well as the resolved
+// IP addresses for each domain. If the node is not configured as
+// an app connector, an error is returned.
+func (b *LocalBackend) handleC2NAppConnectorDomainRoutesGet(w http.ResponseWriter, r *http.Request) {
+	b.logf("c2n: GET /appconnector/routes received")
+
+	if b.appConnector == nil {
+		http.Error(w, "node is not configured as an app connector", http.StatusBadRequest)
+		return
+	}
+
+	var res tailcfg.C2NAppConnectorDomainRoutesResponse
+	res.Domains = b.appConnector.DomainRoutes()
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(res)
```

Comment:
This is fine as written but could more simply be:

```go
w.Header().Set("Content-Type", "application/json")
json.NewEncoder(w).Encode(tailcfg.C2NAppConnectorDomainRoutesResponse{
   Domains: b.appConnector.DomanRoutes()
})
```

(as long as, as noted elsethread, we know the value won't be modified by any other goroutines while this is going on)